### PR TITLE
migrate sbt 1.x

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

**Comment by johnynek** [comment]
looks like my attempt to merge broke everything.

I'm not sure why people let me run projects when I know so little about sbt.

**Comment by ryan-williams** [comment]
I'm guessing your last comment was meant for #213 @johnynek ?

**Comment by johnynek** [comment]
@ryan-williams no, I tried to merge the conflict between here and #213 (which I merged). My attempt to do so failed, making the CI read for this PR that had previously been green (but conflicted with master).

**Comment by ryan-williams** [comment]
Ah I see, the earlier build "failed" due to codecov but passed tests. 

It seems like the latest build passed tests but failed MiMa due to the cats upgrade.

I'm trying to figure out how to run the MiMa checks locally atm to mess with it…

File: .travis.yml [hunk 1]
```diff
@@ -23,23 +23,22 @@ before_install:
...
   - $HOME/.sbt/launchers
   - $HOME/.ivy2/cache
   - $HOME/.nvm
 
 script:
-  - ./sbt ++$TRAVIS_SCALA_VERSION clean coreJS/test lawsJS/test
-  - ./sbt ++$TRAVIS_SCALA_VERSION clean coverage coreJVM/test lawsJVM/test coverageReport
-  - ./sbt ++$TRAVIS_SCALA_VERSION docs/makeMicrosite
+  - ./sbt "++${TRAVIS_SCALA_VERSION}!" clean coreJS/test lawsJS/test
+  - ./sbt "++${TRAVIS_SCALA_VERSION}!" clean coverage coreJVM/test lawsJVM/test coverageReport mimaReportBinaryIssues
```
Comment (ryan-williams) [inline .travis.yml]: @johnynek this is why the build is failing now I think! #213 removed it due to the cats upgrade, but we're going to have to leave it out in any PRs that include that, until there's a new release to compare against

Comment (johnynek) [reply .travis.yml]: Yep. I was the one who said we needed to remove but then manually added it back like a fool.

**Comment by johnynek** [comment]
Thanks y’all. I’ll merge and publish 1.0.0 tomorrow if nothing stops me.

**Comment by johnynek** [comment]
Thank you @xuwei-k !

**Pull request merged**